vnext/Microsoft.ReactNative/Views/TextViewManager.cpp

```diff
@@ -128,6 +130,33 @@ class TextShadowNode final : public ShadowNodeBase {
     this->GetView().as<xaml::Controls::TextBlock>().TextHighlighters().Append(newHigh);
   }
 
+  void ToggleTouchEvents(XamlView xamlView, bool selectable) {
+    if (selectable) {
+      if (m_touchEventHandler == nullptr) {
+        m_touchEventHandler = std::make_unique<TouchEventHandler>(GetViewManager()->GetReactContext());
+      }
+
+      std::function<bool()> shouldCancelOnCaptureLost = [=]() {
```

Comment:
This should be `auto` instead of `std::function<bool()>`.

`auto` resolves to a compiler specific type with the right sized capture group statically allocated. `std::function` is a wrapper which adds a surprising amount of inefficiency on top of it.

Reply:
Oh, I see how you are using this now.

Consider using something like `winrt::event`, which allows delegate style event registration. This is often cleaner than passing callbacks manually.

Reply:
So are you suggesting that we add an event to TouchEventHandler instead of passing in the callback?

Reply:
We could add something like a CaptureLost event to TouchEventHandler with a mutable handled property...

Reply:
Here's what this might look like: https://github.com/rozele/react-native-windows/pull/7/

@NickGerleman - let me know if you'd prefer that I switch to this approach.

Reply:
I honestly don't remember what I was thinking here...

Lmk when you think this is ready for merge

Reply:
@NickGerleman I liked the winrt::event approach better. Passing the callback through was kind of gross. I also simplified a few things and thanks to the `m_rootTag` property on the `ShadowNode`, we can pretty easily pass through the required root view (for setting `pageX` and `pageY` on the touch events) to `TouchEventHandler`, rather than traversing and caching in `TouchEventHandler` :). I think this is ready for another review pass / merge.